app/src/main/java/com/luc4n3x/levyra/feature/dearrow/DeArrowRepository.kt

```
@@ -0,0 +1,107 @@
+    fun close() {
+        scope.cancel()
+        inFlight.clear()
+        synchronized(cache) { cache.clear() }
+    }
+
+    private suspend fun fetchAndCache(videoId: String): DeArrowResult? {
+        val branding = try {
+            brandingSource(videoId)
+        } catch (error: CancellationException) {
+            throw error
+        } catch (error: Throwable) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)